### PR TITLE
Check Command to check completeness of a component version graph

### PR DESCRIPTION
**Pull request by mandelsoft**
### ## Description

The new command `ocm check componentversion` checks component versions in an OCM
repository to be complete. All directly or indirectly references versions must also be 
available in the repository of the initially checked one.

Additionally, it checks for cycles and may be more consistency conditions in the future.

## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [x] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [x] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents

- Related Issue # (issue)
- Closes # (issue)
- Fixes # (issue)
> Remove if not applicable

## Added tests?

- [x] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help
- [ ] Separate ticket for tests # (issue/pr)

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

## Added to documentation?

- [ ] 📜 README.md
- [ ] 🙅 no documentation needed

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by ikhandamirov** [comment]
@mandelsoft, please see my question in #518

File: cmds/ocm/commands/ocmcmds/components/check/cmd.go [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+		utils.Names(Names, names...)...,
+	)
+}
+
+func (o *Command) ForName(name string) *cobra.Command {
+	return &cobra.Command{
+		Use:   "[<options>] {<component-reference>}",
+		Short: "check completeness of a component version in an OCM repository",
+		Long: `
+This command checks, whether component versiuons are completely contained
```
Comment (hilmarf) [inline cmds/ocm/commands/ocmcmds/components/check/cmd.go:60]:
```suggestion
This command checks, whether component versions are completely contained
```

File: docs/reference/ocm.md [hunk 2]
```diff
@@ -11,7 +11,7 @@ ocm [<options>] <sub command> ...
 ```
   -X, --attribute stringArray     attribute setting
       --ca-cert stringArray       additional root certificate authorities
-      --config string             configuration file
+      --config stringArray        configuration file
```
Comment (hilmarf) [inline docs/reference/ocm.md:14]: Will the client still be backwards compatible?

File: docs/reference/ocm.md [hunk 3]
```diff
@@ -318,8 +340,9 @@ by a certificate delivered with the signature.
 
 ##### Sub Commands
 
-* [ocm <b>add</b>](ocm_add.md)	 &mdash; Add resources or sources to a component archive
+* [ocm <b>add</b>](ocm_add.md)	 &mdash; Add elements to a component repository or component version
 * [ocm <b>bootstrap</b>](ocm_bootstrap.md)	 &mdash; bootstrap components
+* [ocm <b>check</b>](ocm_check.md)	 &mdash; check components in OCM repository
```
Comment (hilmarf) [inline docs/reference/ocm.md:345]:
```suggestion
* [ocm <b>check</b>](ocm_check.md)	 &mdash; Check components in OCM repository
```

File: pkg/contexts/credentials/utils.go [hunk 4]
```diff
@@ -28,3 +32,63 @@ func CredentialsFor(ctx ContextProvider, obj interface{}, uctx ...UsageContext)
...
+					min = d
+				}
+			}
+		}
+		if fix != "" {
+			return fix
+		}
+	}
+	return spec
+}
```
Comment (hilmarf) [inline pkg/contexts/credentials/utils.go:94]: Cool and nice feature, but not related to `check` command... wouldn't a separate PR make sense?

File: pkg/errors/errprop.go [hunk 5]
```diff
@@ -22,3 +26,10 @@ func PropagateErrorf(errp *error, f ErrorFunction, msg string, args ...interface
 		*errp = ErrListf(msg, args...).Add(*errp, f()).Result()
 	}
 }
+
+func LogError(log logging.Logger, f ErrorFunction, msg string, keypair ...interface{}) {
+	err := f()
+	if err != nil {
+		log.LogError(err, msg, keypair...)
+	}
+}
```
[the remark this answers is not in the document]
Comment (mandelsoft) [reply pkg/errors/errprop.go:35]: jow, this is the function you requested to log errors from defer functions. I just added it, here

File: docs/reference/ocm.md [hunk 6]
```diff
@@ -225,6 +225,28 @@ The value can be a simple type or a JSON/YAML string for complex values
...
+         "path": ""&lt;file path>"
+       }
+    ],
+  </pre>
+
+  One of following data fields are possible:
+  - <code>data</code>:       base64 encoded binary data
+  - <code>stringdata</code>: plain text data
+  - <code>path</code>:       a file path to read the data from
+
```
[the remark this answers is not in the document]
Comment (mandelsoft) [reply docs/reference/ocm.md:249]: generated content. Probably not regenerated by another PR

File: cmds/ocm/commands/ocmcmds/components/check/cmd.go
Comment (hilmarf) [inline cmds/ocm/commands/ocmcmds/components/check/cmd.go:1]: A few more comments at the coding would be great to easier understand what's going on... e.g. GoDoc at function

**Pull request merged**